Week6/Battery.tsx

```diff
@@ -0,0 +1,106 @@
+import { useEffect, useState, useCallback } from "react";
+import * as Battery from "expo-battery";
+import { StyleSheet, Text, View } from "react-native";
+import { AccelerometerSDK } from "./Accelerometer";
+
+export const BatterySDK = () => {
+  const [batteryLevel, setBatteryLevel] = useState<number | null>(null);
+  const [subscription, setSubscription] = useState<Battery.Subscription | null>(
+    null
+  );
+
+  const _subscribe = async () => {
+    const batteryLevel = await Battery.getBatteryLevelAsync();
+    setBatteryLevel(batteryLevel);
+  };
+
+  const _unsubscribe = useCallback(() => {
+    subscription && subscription.remove();
+    setSubscription(null);
+  }, [subscription]);
+
+  useEffect(() => {
+    _subscribe();
+    return () => _unsubscribe();
+  }, [_unsubscribe]);
+
+  const { x, y, z } = AccelerometerSDK();
+  const magnitude = Math.sqrt(x * x + y * y + z * z);
+
+  useEffect(() => {
+    console.log("Magnitude:", magnitude);
+
+    if (magnitude > 1.1) {
+      console.log("Updating battery level...");
+      setBatteryLevel((batteryLevel) =>
+        Math.min(batteryLevel as any + 0.01, 1)
```

Comment:
casting this as `any` only hides the problem. the variable is still nullable so it can still cause a crash. Instead of initializing `batteryLevel` to null, set it to 0